Add seven new Azure Functions templates

BigMorty opened the pull request:
Please fill out this template! There are three different types of contributions, feel free to delete the checklists that are not applicable to your contribution type. 

# If you are submitting a new `azd` template to the gallery
Fill this out if you want your template to be added to the awesome-azd gallery!

### Your template repository
Place your template repository link here:
https://github.com/Azure-Samples/functions-quickstart-dotnet-azd
https://github.com/Azure-Samples/functions-quickstart-python-http-azd
https://github.com/Azure-Samples/functions-quickstart-javascript-azd
https://github.com/Azure-Samples/functions-quickstart-typescript-azd
https://github.com/Azure-Samples/azure-functions-java-flex-consumption-azd
https://github.com/Azure-Samples/functions-quickstart-powershell-azd
https://github.com/Azure-Samples/Azure-Functions-OpenAI-Demo

<!-- Please ensure that your PR includes the following metadata! All fields are mandatory unless explicitly marked as optional -->

- [x] Added an entry to https://github.com/Azure/awesome-azd/website/static/templates.json that includes:
    - [x] **Template title** - A short title that reflects the local application stack that someone could use to get their application on Azure (e.g. "Containerized React Web App with Java API and MongoDB")
    - [x] **Description** - 1-2 sentence description of the architecture (e.g. Azure services) or solution that is defined by the template.
    - [x] **Architecture Diagram or Application Screenshot** - The image should include all services and their connections ([example](https://github.com/Azure-Samples/todo-csharp-sql/blob/main/assets/resources.png)). You should add the image to the `website/src/data/images/`.
    - [x] **Link to Author's GitHub or other relevant website** - Used for attribution
    - [x] **Author's Name** -  Name to credit on the gallery card
    - [x] **Link to template source** - Link to the template GitHub repo 
    - [x] **Tags** - Specify [tags](https://github.com/Azure/awesome-azd/blob/main/website/src/data/tags.tsx) to represent the template. If you don't see a relevant tag for your template? Feel free to add one!
        
        **Required tags**:
      - [x] Tag your template as Micrsoft-authored (`"msft"`) or Community-authored (`"community"`)
      - [x] Tag the IaC provider (`"bicep"` or `"terraform"`)
      - [x] At least 1 tag for programming language used
      - [x] At least 1 tag for Azure services integrated
      - [x] Add the `"new"` tag for any newly authored templates

- [ ] In the PR comment, if you can also add a link to the PR where you made your repo `azd` compatible this will allow us to provide feedback on your template and speed up the review process.
- [ ] If the template is Microsoft-authored, we encourage you to also [publish it to learn.microsoft.com/samples](https://review.learn.microsoft.com/en-us/help/contribute/samples/process/onboarding?branch=main).

# If you are submitting a resource to be added to the awesome-azd README:
- [ ] Name of resource
- [ ] Resource author - who created this resource? (so we can credit them!)
- [ ] What section should this resource be included in? -Is the resource an article? A video? Something else?

<!-- Once submitted, the issue will be reviewed - we plan to do reviews on a rolling basis at regular intervals. The process will include verifying all information required for the template gallery is provided and the template works (i.e., successfully deploys to Azure with `azd up`). 
 * If we have questions or enhancements, we will add comments in issue thread (**issue stays open**)
 * If the contribution is approved, we'll merge the PR to update the gallery (**issue will then be closed**) -->

vhvb1989 commented on website/static/templates.json:
```diff
@@ -2194,5 +2194,68 @@
+    "website": "https://github.com/Azure-Samples",
+    "author": "Azure Functions Team",
+    "source": "https://github.com/Azure-Samples/functions-quickstart-dotnet-azd",
+    "tags": ["new", "functions", "flex", "dotnetCsharp", "http", "entra-id", "vnet", "appinsights", "msft", "bicep"]
```
Make sure the tags you are using exists here: https://github.com/Azure/awesome-azd/blob/main/website/src/data/tags.tsx#L135

For example, change `vnet` to `vnets` :https://github.com/Azure/awesome-azd/blob/main/website/src/data/tags.tsx#L512

Remove `flex` or add a new tag `flex` to the tags file.

BigMorty replied:
Fixed

BigMorty commented:
@vhvb1989, tags have been updated. thanks

v-xuto commented:
> @v-xuto Could you test these templates please?

Sure. Testing.

v-xuto commented:
@hemarina, @BigMorty  We have finished this template test, and we filed six issues. Please review.
- https://github.com/Azure-Samples/azure-functions-java-flex-consumption-azd/issues/8
- https://github.com/Azure-Samples/azure-functions-java-flex-consumption-azd/issues/9
- https://github.com/Azure-Samples/functions-quickstart-powershell-azd/issues/4
- https://github.com/Azure-Samples/Azure-Functions-OpenAI-Demo/issues/17
- https://github.com/Azure-Samples/Azure-Functions-OpenAI-Demo/issues/18
- https://github.com/Azure-Samples/Azure-Functions-OpenAI-Demo/issues/19

vhvb1989 commented:
@BigMorty , 

The tag "http" is not defined in ./src/data/tags.tsx.

BigMorty commented:
Thanks, I will work on fixing these issues and the tag.

BigMorty commented:
@v-xuto @hemarina 
- I could not repo the F5 issue, F5 worked fine
- I fixed the links in the two Readme files
- I removed the AI template from the PR, we are going to refactor that into three simpler templates instead
- I removed the http tag, it is not really needed

v-xuto commented:
@BigMorty The following are the latest status of each issue:
- https://github.com/Azure-Samples/azure-functions-java-flex-consumption-azd/issues/8 -> **F5 issue, we have added the latest comments, please review.**
- https://github.com/Azure-Samples/azure-functions-java-flex-consumption-azd/issues/9 -> **Fixed, have verified.**
- https://github.com/Azure-Samples/functions-quickstart-powershell-azd/issues/4 -> **The first issue was fixed, the second one was not. Please review.**
- https://github.com/Azure-Samples/Azure-Functions-OpenAI-Demo/issues/17 -> **Ignore, because you removed the AI template from this PR. If you need to close this issue, please let me know.**
- https://github.com/Azure-Samples/Azure-Functions-OpenAI-Demo/issues/18 -> **Fixed, have verified.**
- https://github.com/Azure-Samples/Azure-Functions-OpenAI-Demo/issues/19 -> **Ignore, because you removed the AI template from this PR. If you need to close this issue, please let me know.**

v-xuto commented:
@hemarina All issues have been fixed (except AI template [issue#17](https://github.com/Azure-Samples/Azure-Functions-OpenAI-Demo/issues/17)). There are currently no issues with this template. Please review.

BigMorty commented:
Thanks all. @hemarina, when can we expect these to show here - https://azure.github.io/awesome-azd/?tags=functions

BigMorty commented:
@hemarina, Wednesday is fine. thanks!
